src/QsCompiler/Tests.Compiler/TestCases/LinkingTests/IntrinsicMapping.qs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
Looking at this, it might make sense to take it as a precondition that the environment tree does not contain any type parametrized callables. What do you think?

Reply:
Validating the state of the incoming environment tree could get pretty involved. I think that should be done separately from this transformation, in whatever process we have for generating those environment trees.